Framework/API/src/ExperimentInfo.cpp

```diff
@@ -446,58 +446,33 @@ T getParam(const std::string &paramType, const std::string &paramValue) {
   return param->value<T>();
 }
 
-void updatePosition(DetectorInfo &detectorInfo, const Instrument &instrument,
-                    const IComponent *component, const V3D &newRelPos) {
-  // Important: Get component WITH ParameterMap so we see correct parent
-  // positions and rotations and DetectorInfo updates correctly.
-  if (!instrument.isParametrized())
-    throw std::runtime_error(
-        "Need parametrized instrument for updating positions");
-  const auto &parComp = instrument.getComponentByID(component);
-  auto position = newRelPos;
-  if (auto parent = parComp->getParent()) {
-    parent->getRotation().rotate(position);
-    position += parent->getPos();
-  }
-  detectorInfo.setPosition(*parComp, position);
-}
-
-void updateRotation(DetectorInfo &detectorInfo, const Instrument &instrument,
-                    const IComponent *component, const Quat &newRelRot) {
-  // Important: Get component WITH ParameterMap so we see correct parent
-  // positions and rotations and DetectorInfo updates correctly.
-  if (!instrument.isParametrized())
-    throw std::runtime_error(
-        "Need parametrized instrument for updating rotations");
-  const auto &parComp = instrument.getComponentByID(component);
-  auto rotation = newRelRot;
-  if (auto parent = parComp->getParent()) {
-    // Note the unusual order. This is what Component::getRotation does.
-    rotation = parent->getRotation() * newRelRot;
-  }
-  detectorInfo.setRotation(*parComp, rotation);
+void updatePosition(ComponentInfo &componentInfo, const IComponent *component,
+                    const V3D &newRelPos) {
+  const auto compIndex = componentInfo.indexOf(component->getComponentID());
+  componentInfo.setPosition(compIndex,
+                            componentInfo.position(compIndex) + newRelPos);
```

Comment:
I'm going to add to `ComponentInfo`/ `InfoComponentVisitor` data allowing us to go from a componentIndex to the componentIndex of the parent. I need this in a few other places too.